Azure-Samples/aks-event-driven-replicate-from-aws
# Karpenter updates

## Purpose
This pull request contains changes made to correct how the application [Karpenter] node pool is created. The changes also include adding a network rule for storage account access in compliance with the policy forbidding public access to Azure Storage accounts.
* ...

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Test by cloning the repo and running the `deploy.sh` script. Verify that the Karpenter node pool was created by running the following command

`kubectl get nodepool`

You should see a node pool named `default`. If so, run the following command:

`kubectl describe nodepool default`

In the output, you should see these lines:

```
APIVersion:        karpenter.sh/v1beta1
Kind:                  NodePool
```

## What to Check
Verify that the following are valid
* ...

## Other Information
